great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1071 @@
+        if not column_type:
+            column_type = self._get_column_type(dataset, column_name)
+            column_info_entry["type"] = column_type
+
+            dataset.set_config_value("interactive_evaluation", True)
+
+        return column_type
+
+    def _get_column_type(self, dataset, column):
+        """
+        Determines the data type of a column by evaluating the success of `expect_column_values_to_be_in_type_list`.
+        In the case of type Decimal, this data type is returned as NUMERIC, which contains the type lists for both INTs
+        and FLOATs.
+
+        The type_list expectation used here is removed, since it will need to be built once the build_suite function is
+        actually called. This is because calling build_suite wipes any existing expectations, so expectations called
+        during the init of the profiler do not persist.
+
+        Args:
+            dataset: A GE dataset
+            column: The column for which to get the data type
+
+        Returns:
+            The data type of the specified column
+        """
+        # list of types is used to support pandas and sqlalchemy
+        type_ = None
+        dataset.set_config_value("interactive_evaluation", True)
+        try:
+
+            if (
+                dataset.expect_column_values_to_be_in_type_list(
+                    column, type_list=sorted(list(ProfilerTypeMapping.INT_TYPE_NAMES))
+                ).success
+                and dataset.expect_column_values_to_be_in_type_list(
+                    column, type_list=sorted(list(ProfilerTypeMapping.FLOAT_TYPE_NAMES))
+                ).success
+            ):
+                type_ = "NUMERIC"
+
+            elif dataset.expect_column_values_to_be_in_type_list(
+                column, type_list=sorted(list(ProfilerTypeMapping.INT_TYPE_NAMES))
+            ).success:
+                type_ = "INT"
+
+            elif dataset.expect_column_values_to_be_in_type_list(
+                column, type_list=sorted(list(ProfilerTypeMapping.FLOAT_TYPE_NAMES))
+            ).success:
+                type_ = "FLOAT"
+
+            elif dataset.expect_column_values_to_be_in_type_list(
+                column, type_list=sorted(list(ProfilerTypeMapping.STRING_TYPE_NAMES))
+            ).success:
+                type_ = "STRING"
+
+            elif dataset.expect_column_values_to_be_in_type_list(
+                column, type_list=sorted(list(ProfilerTypeMapping.BOOLEAN_TYPE_NAMES))
+            ).success:
+                type_ = "BOOLEAN"
+
+            elif dataset.expect_column_values_to_be_in_type_list(
+                column, type_list=sorted(list(ProfilerTypeMapping.DATETIME_TYPE_NAMES))
+            ).success:
+                type_ = "DATETIME"
+
+            else:
+                dataset.expect_column_values_to_be_in_type_list(column, type_list=None)
+                type_ = "UNKNOWN"
+        except NotImplementedError:
+            type_ = "unknown"
+
+        if type_ == "numeric":
```

Comment:
See above comment about using the enum value e.g.:
```suggestion
        if type_ == ProfilerDataType.NUMERIC.value:
```
That way you don't have to worry about making sure it's the right case either. In this case if you don't decide to use the enum value I think it should be "NUMERIC" to match the case above on line L393 https://github.com/great-expectations/great_expectations/pull/2232/files#diff-3c8f91aa754f6a5468b978ffeb0e73350ce6446b29345026f71d6a7261524c3fR393

Reply:
Thanks! I decided to not use the enum, but I did convert all of the casing for types and cardinality to upper so as to be consistent.